docs/articles/documentation/test-api/actions/upload.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: docs
+title: Upload
+permalink: /documentation/test-api/actions/upload.html
+---
+# Upload
+
+Use the following actions to work with file upload input elements.
+
+* [Populate File Upload Input](#populate-file-upload-input)
+* [Clear File Upload Input](#clear-file-upload-input)
+
+## Populate File Upload Input
+
+Populates the specified file upload input with file paths.
+
+```text
+setFilesToUpload( selector, filePath )
+setFilesToUpload( selector, [filePath, ...] )
+```
+
+Parameter  | Type                                              | Description
+---------- | ------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------
+`selector` | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies the input field to which file paths are written. See [Selecting Target Elements](index.md#selecting-target-elements).  
+`filePath` | String                                            | The path to the uploaded file. Relative paths resolve from the folder with the test file.
+
+The following example illustrates how to use the `setFilesToUpload` action.
+
+```js
+test('Uploading', async t => {
+    await t
+        .setFilesToUpload('#upload-input', [
+            './uploads/1.jpg',
+            './uploads/2.jpg',
+            './uploads/3.jpg'
+        ])
+        .click('#upload-button');
+});
+```
+
+The `setFilesToUpload` action removes all file paths from the input before populating it with new ones.
+
+> If an error occurs while uploading files, the test will fail.
+
+## Clear File Upload Input
+
+Removes all file paths from the specified file upload input.
+
+```text
+clearUpload( selector )
+```
+
+Parameter  | Type                                              | Description
+---------- | ------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------
+`selector` | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies the input field that needs to be cleared. See [Selecting Target Elements](index.md#selecting-target-elements).  
+
+The example below shows how to use the `clearUpload` action.
+
+```js
+import { expect } from 'chai';
+
+test('Trying to upload with no files specified', async t => {
+    await t
+        .clearUpload('#upload-input')
+        .setNativeDialogHandler(() => true)
```

Comment:
I think we can get rid of native dialogs handling in this example to simplify it. What do you think?